models/flexload_milp.py

```diff
@@ -594,46 +617,57 @@ def solve_relaxedproblem(self, tee=False):
-                "The solution violates the bounds of flexible operation."
-                "\nResolving the problem with increased bound penalties."
+        # print the continuous load bounds
+        print(
+            f"Continuous load bounds: [{lower_bound:.2f}, {upper_bound:.2f}] "
+            f"with average continuous load: {cont_load_avg:.2f}"
+        )
+        # print the continuous load
+        print(f"Max load: {np.max(cont_load):.2f}, Min load: {np.min(cont_load):.2f}")
+
+        resolve = 0
+
+        # check min continuous load bound
+        if lower_bound - self.model.min_contload.value > self.tol:
+            resolve = 1
+            self.model.min_contload_penalty.value = (
+                self.model.min_contload_penalty.value + descent_stepsize * (cont_load_avg - self.model.min_contload.value)
             )
-            # raise ValueError("Continuous load violated bounds of relaxed problem.")
-
-            # TODO - think about implementing a smarter algorithm for heuristic bound enforcement
-            stepsize = 1e-5
-
-            while (
-                np.max(np.abs(cont_load - cont_load_avg)) - self.tol
-                >= flex_capacity * cont_load_avg
-            ):
-                # get the violation of the maximum continuous load
-                max_contload_violation = np.max(
-                    [(1 + flex_capacity) * cont_load_avg - np.max(cont_load), 0]
-                )
-                self.model.max_contload_penalty.value = (
-                    self.model.max_contload_penalty.value
-                    + max_contload_violation * stepsize
-                )
+        else:
+            pass
 
-                # get the violation of the minimum continuous load
-                min_contload_violation = np.max(
-                    [np.min(cont_load) - (1 - flex_capacity) * cont_load_avg, 0]
-                )
-                self.model.min_contload_penalty.value = (
-                    self.model.min_contload_penalty.value
-                    + min_contload_violation * stepsize
-                )
+        if self.model.max_contload.value - upper_bound > self.tol:
+            resolve = 1
+            self.model.max_contload_penalty.value = (
+                self.model.max_contload_penalty.value + descent_stepsize * (self.model.max_contload.value - cont_load_avg)
+            )
+        else:
+            pass
 
-                # update the model and call solve again
-                results = solver.solve(self.model, tee=tee)
+        if resolve == 0:
+            # if the solution is feasible, return it
+            print("The solution is feasible within the provided bounds.")
+            return self.model, results
 
+        elif recursion_counter < max_iter-1 and resolve > 0:
+            recursion_counter += 1
+            print(
+                f"The solution violated the bounds of flexible operation.\nResolving the problem with increased bound penalties.\nTry: {recursion_counter+1}"
```

Comment:
nit: to avoid overly long lines I would take advantage of Python's automatic string concatenation and break this up as follows:

```python
print(
    "The solution violated the bounds of flexible operation.\n"
    "Resolving the problem with increased bound penalties.\n"
    f"Try: {recursion_counter+1}"
)
```